src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -0,0 +1,118 @@
+(ns puppetlabs.trapperkeeper.services.authorization.authorization-core
+  (:require [clojure.string :as str]
+            [puppetlabs.kitchensink.core :as ks]
+            [schema.core :as schema]
+            [puppetlabs.trapperkeeper.authorization.rules :as rules]
+            [puppetlabs.trapperkeeper.authorization.acl :as acl])
+  (:import (java.util.regex PatternSyntaxException)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Constants
+
+(def required-keys
+  "Keys required in an auth rule map."
+  [:path :type])
+
+(def required-or-key
+  "At least one of these keys is required in an auth rule map."
+  [:deny :allow])
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Private
+
+(defn pprint-rule
+  [rule]
+  (str/trim (ks/pprint-to-string rule)))
+
+(schema/defn transform-config-rule-to-acl :- acl/ACL
+  [{:keys [allow deny]}]
+  (let [allow (if (string? allow) [allow] allow)
+        deny (if (string? deny) [deny] deny)]
+    (let [allow-acl (reduce #(acl/allow %1 %2) acl/empty-acl allow)
+          full-acl (reduce #(acl/deny %1 %2) allow-acl deny)]
+      full-acl)))
+
+(schema/defn transform-config-rule :- rules/Rule
+  [config-rule]
+  (let [type (name (.toLowerCase (:type config-rule)))]
+    {:type (if (= type "regex") :regex :string)
+     :path (re-pattern (:path config-rule))
+     :method :any
+     :acl (transform-config-rule-to-acl config-rule)}))
+
+(defn validate-auth-config-rule!
```

Comment:
Just as a general comment, I think it would be cleaner to maintain this code if we were using Prismatic Schema to do the heavy lifting of the validations rather than all of the hand-rolled stuff that is here.  That said, though, I understand how onerous it is for a user to interpret the error messages from a Schema validation failure and that the code here may be an attempt to compensate for those and create much more readable failures for users.  I wouldn't block the PR based on this, but it's probably worth revisiting at some point later on.  As it is, I expect that we'll have a fair bit of turmoil here in follow-on PRs since the format of the rule definitions are still a little in flux.